pkg/upload/batch_test.go

```diff
@@ -65,3 +66,29 @@ func captureOutput(f func()) string {
 	log.SetOutput(os.Stderr)
 	return buf.String()
 }
+
+func TestWaitWithNewFiles(t *testing.T) {
+	group := file.NewGroup("package.json", nil, []string{"yarn.lock"})
+	var groups file.Groups
+	groups.Add(*group)
+	metaObj, err := git.NewMetaObject("", "repository-name", "commit-name", "", "", "")
+	if err != nil {
+		t.Fatal("failed to create new MetaObject")
+	}
+
+	var c client.IDebClient
+	clientMock := testdata.NewDebClientMock()
+	mockRes := testdata.MockResponse{
+		StatusCode:   http.StatusCreated,
+		ResponseBody: io.NopCloser(strings.NewReader("{}")),
+	}
+	clientMock.AddMockResponse(mockRes)
+	c = clientMock
+	batch := newUploadBatch(&c, groups, metaObj, "CLI")
+
+	uploadResult, err := batch.wait()
+
+	if uploadResult != nil && err != nil {
+		t.Fatal("Upload result and err must be nil")
+	}
```

Comment:
Nice test! :tada: 
I am just a bit worried about one thing. What happens  if `uploadResult` is `nil` but `err` is not? 
Then the test will pass when it should fail.
I would do something like this:
```suggestion
	if uploadResult != nil || err != nil {
		t.Fatal("Upload result and err must be nil")
	}
```

Reply:
In my version of the code uploadResult and err must be nil at the same time to pass the test, so the test will fail when it should fail.

Reply:
Alright. Will the test pass if you change the logic back to how it was? My point is that you miss the case when err is not nil